lib/engine.js

```diff
@@ -17,80 +17,93 @@
 \*───────────────────────────────────────────────────────────────────────────*/
 'use strict';
 
+var freshy = require('freshy').freshy;
 var path = require('path'),
-    dust = require('dustjs-linkedin'),
     utils = require('./utils'),
-    patch = require('./patch'),
     River = require('./brook'),
     reqwire = require('./reqwire');
+var dustjacket = require('dustjacket');
+var fs = require('fs');
 
 
-var loaders = {
-    dust: require('./reader/dust'),
-    js: require('./reader/js')
-};
+function loader (name, context, cb) {
+    var views = utils.resolveViewDir(context.global);
 
+    var ext = path.extname(name) || 'dust';
+    if (ext[0] !== '.') {
+        ext = '.' + ext;
+    }
 
+    var file = path.resolve(views, name + ext);
+    fs.readFile(file, 'utf8', function (err, content) {
+        if (err) {
+            if (err.code !== 'ENOENT') {
+                cb(err);
+            } else {
+                cb();
+            }
+        } else {
+            cb(null, content );
+        }
+    });
+}
 
-function createRenderContext(options) {
-    var context;
+exports.create = function (config) {
+    config = config || {};
 
-    // Hoist *only* view, settings, and ext to global context.
-    context = dust.makeBase({
-        views: options.views,
-        settings: options.settings,
-        ext: options.ext
-    });
+    var dust = freshy('dustjs-linkedin', function (dust) {
+        require('dustjs-helpers');
```

Comment:
These two lines give me a little anxiety without more context. I'm wondering if some comments will help our future selves remember that the callback is intentionally synchronous to allow some work to be done (most likely manipulation of global state WRT `require('dustjs-helpers');`) prior to the freshly-loaded module being unloaded and ultimately returned.